src/wp-includes/rest-api/endpoints/class-wp-rest-global-styles-controller.php

```diff
@@ -41,7 +41,11 @@ public function register_routes() {
 		// List themes global styles.
 		register_rest_route(
 			$this->namespace,
-			'/' . $this->rest_base . '/themes/(?P<stylesheet>[\/\s%\w\.\(\)\[\]\@_\-]+)',
+			/**
+			 * Matches the theme's directory: `/themes/<subdirectory>/<theme>/` or `/themes/<theme>/`.
+			 * Note: `|` is an invalid character for a Windows' directory name.
+			 */
+			'/' . $this->rest_base . '/themes/(?P<stylesheet>[^\/\|]+(?:\/[^\/\|]+)?)',
```

Comment:
Before [52376](https://core.trac.wordpress.org/changeset/52376), the regex here was this:
```php
'/' . $this->rest_base . '/themes/(?P<stylesheet>[^.\/]+(?:\/[^.\/]+)?)'
```

It was excluding `.` from being matched. The new regex removed the `.` and added `|` (as it's an invalid character on Windows). Bam, that works for all valid directory name characters including `é`, `<>`, `=`, etc.